### DISCUSSION TITLE
HDM should evaluate YAML data

### DISCUSSION THREAD

**Discussion by tuxmea** [betadots/hdm#480] [Ideas]
- duplicate keys (on top level)
- duplicate keys (in hash)

**Comment by bastelfreak** [comment]
puppet-syntax is usually used in pipelines to validate data in modules: https://github.com/voxpupuli/puppet-syntax/blob/master/lib/puppet-syntax/hiera.rb

It currently has no check for duplicate keys, I think it should be added there.
